File: 1652-DefuseBomb.py
class Solution(object):
    def decrypt(self, code, k):
        """
        :type code: List[int]
        :type k: int
        :rtype: List[int]
        """
        output=[]
        if k==0:
            return [0]*len(code)
        if k>0:
            for i in range(len(code)):
                sum=0
                d=i+1
                for j in range(k):
                    d= d%len(code)
                    sum+=code[d]
                    d+=1
                output.append(sum)
                sum=0
        else:
            for i in range(len(code)):
                sum=0
                d=i-1
                for j in range(abs(k)):
                    d= d%len(code)
                    sum+=code[d]
                    d-=1
                output.append(sum)
                sum=0
        
        return output



        

code = [2,4,9,3]
k = -2
sol=Solution()
print(sol.decrypt(code,k))

File: 1652-DefuseBomb.java
class Solution {
    public int[] decrypt(int[] code, int k) {
    
    int[] mycode= new int[code.length];
    int windowSum=0;
    int n=code.length;
    if(k>0){
       for(int i=1; i<=k;i++){
        windowSum=windowSum+code[i%n];
       }
        for(int i=0; i<code.length;i++){
            mycode[i]=windowSum;
            windowSum=windowSum-code[(i+1)%n]+code[(i+k+1)%n];         
        }
    }
    else if(k<0){
        k=-k;
       for(int i=1; i<=k;i++){
        windowSum=windowSum+code[(n-i)%n];
       }
        for(int i=0; i<code.length;i++){
            mycode[i]=windowSum;
            // windowSum=windowSum-code[(n-i-1)%n]+code[(n-i-1-k+n)%n]; 
           windowSum = windowSum - code[(n + i - k) % n] + code[i % n];


        }
    }

    return mycode;    
    }
}